src/main/java/io/cdap/plugin/gcp/bigquery/sink/BigQuerySinkConfig.java

```diff
@@ -172,4 +179,44 @@ private void validateColumnForPartition(@Nullable String columnName, @Nullable S
                                                     columnName, type));
     }
   }
+
+  private void validateClusteringOrder(@Nullable Schema schema) {
+    if (!shouldCreatePartitionedTable() || Strings.isNullOrEmpty(clusteringOrder) || schema == null) {
+      return;
+    }
+    List<String> columnsNames = Arrays.stream(clusteringOrder.split(",")).map(String::trim)
+      .collect(Collectors.toList());
+    if (columnsNames.size() > MAX_NUMBER_OF_COLUMNS) {
+      throw new InvalidConfigPropertyException(
+        String.format("Expected no more than '%d' clustering fields, found '%d'.", MAX_NUMBER_OF_COLUMNS,
+                      columnsNames.size()), NAME_CLUSTERING_ORDER);
+    }
+    for (String column : columnsNames) {
+      Schema.Field field = schema.getField(column);
+      if (field == null) {
+        throw new InvalidConfigPropertyException(
+          String.format("Clustering column '%s' is missing from the table schema", column),
```

Comment:
is this the table schema or the input schema?

Reply:
Output schema will be used if it not null. In other case - will be used input schema.

Reply:
let's say 'input schema' instead of 'table schema' then, since it's coming from the pipeline and not from bigquery.